app/controllers/application_controller.rb

```diff
@@ -37,15 +37,15 @@ def reject_without_key!
     end
   end
 
+  delegate :prison_name, :visit_id, to: :visit, allow_nil: true
+
   def add_extra_sentry_metadata
     response['X-Request-Id'] = request_id
-    Raven.extra_context(request_id: request_id)
-    if visit = session[:visit]
-      Raven.extra_context(visit_id: visit.visit_id)
-      if prison = visit.prisoner.prison_name
-        Raven.extra_context(prison: prison)
-      end
-    end
+    {
+      request_id: request_id,
+      visit_id: visit_id,
+      prison: prison_name
+    }.map { |key, value| Raven.extra_context(key => value) if value }
```

Comment:
I don't know what Raven is, but it looks like it has side effects. If so, perhaps `each do … end` might be more intention-revealing